packages/teleterm/src/ui/services/workspacesService/workspacesService.test.ts

```diff
@@ -0,0 +1,115 @@
+import { Workspace, WorkspacesService } from './workspacesService';
+import { ClustersService } from '../clusters';
+import { StatePersistenceService } from '../statePersistence';
+
+describe('restoring workspace', () => {
+  function getTestSetup(options: {
+    clusterUri: string; // assumes that only one cluster can be added
+    persistedWorkspaces: Record<string, Workspace>;
+  }) {
+    const statePersistenceService: Partial<StatePersistenceService> = {
+      getWorkspacesState: () => ({
+        workspaces: options.persistedWorkspaces,
+      }),
+      saveWorkspacesState: jest.fn(),
+    };
+
+    const clustersService: Partial<ClustersService> = {
+      subscribe: jest.fn(),
+      unsubscribe: jest.fn(),
+      findRootClusterByResource: jest.fn(),
+      findCluster: jest.fn(),
+      findGateway: jest.fn(),
+      getClusters: () => [
+        {
+          uri: options.clusterUri,
+          name: 'Test cluster',
+          connected: true,
+          leaf: false,
+          proxyHost: 'test:3030',
+          actualName: 'Test cluster',
+          loggedInUser: {
+            name: 'Alice',
+            rolesList: [],
+            sshLoginsList: [],
+          },
+        },
+      ],
+    };
+
+    const clusterDocument = {
+      kind: 'doc.cluster',
+      title: 'Cluster Test',
+      clusterUri: options.clusterUri,
+      uri: 'docs/test-cluster-uri',
+    };
+
+    const workspacesService = new WorkspacesService(
+      undefined,
+      // @ts-expect-error using mocks
+      clustersService,
+      undefined,
+      statePersistenceService
+    );
+
+    workspacesService.getWorkspaceDocumentService = () => ({
+      // @ts-expect-error using mocks
+      createClusterDocument() {
+        return clusterDocument;
+      },
+    });
```

Comment:
That's fine for now, but I think we might be able to get rid of `@ts-expect-error` by mocking stuff through Jest. I encourage you to give it a try next time, see `packages/teleterm/src/ui/services/quickInput/quickInputService.test.ts` for an example.

Reply:
Thanks, I'll do it next time!